### PR TITLE
Add retries for video encode to succeed

### PR DESCRIPTION
**Pull request by p-tsai**

### CODE REVIEW

File: videokit/src/main/java/com/groupme/android/videokit/VideoTranscoder.java [hunk 1]
```diff
@@ -514,12 +515,22 @@ private boolean extractAndFeedDecoder(MediaCodec decoder, ByteBuffer[] buffers,
...
-                    0,
-                    MediaCodec.BUFFER_FLAG_END_OF_STREAM);
+            try {
+                decoder.queueInputBuffer(
+                        decoderInputBufferIndex,
+                        0,
+                        0,
+                        0,
+                        MediaCodec.BUFFER_FLAG_END_OF_STREAM);
+            } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (p-tsai) [reply videokit/src/main/java/com/groupme/android/videokit/VideoTranscoder.java]: Seems that buffer has not actually hit the end of the input (there's more input data remaining) but `extractor` thinks it has reached end of stream. So sending the `BUFFER_FLAG_END_OF_STREAM` when there's still remaining data raises this exception. Also tried `decoder.signalEndOfInputStream()` here which got an illegalStateException.  


https://stackoverflow.com/questions/48919712/mediacodec-strange-error-when-sending-eos-buffer-into-audio-decoder/50678479

[the remark this answers is not in the document]
Comment (p-tsai) [reply videokit/src/main/java/com/groupme/android/videokit/VideoTranscoder.java]: Probably due to this: https://developer.android.com/reference/android/media/MediaExtractor.html#advance()
 

> `public boolean advance ()`
Advance to the next sample. Returns false if no more sample data is available (end of stream). When extracting a local file, the behaviors of advance() and readSampleData(ByteBuffer, int) are undefined in presence of concurrent writes to the same local file; more specifically, end of stream could be signalled earlier than expected.

[the remark this answers is not in the document]
Comment (mwdavis84) [reply videokit/src/main/java/com/groupme/android/videokit/VideoTranscoder.java]: Makes sense. Can we just catch the IllegalStateException instead, then?

[the remark this answers is not in the document]
Comment (p-tsai) [reply videokit/src/main/java/com/groupme/android/videokit/VideoTranscoder.java]: actually the exception here would be android.media.mediacodec$codecexception which has an API level warning. should I still just catch for `MediaCodec.CodeException` despite the warning? 

(`IllegalStateException` came up when I was trying something else :) )

**Pull request merged**